components/b2b/AccountHierarchy/AccountHierarchyEditForm/AccountHierarchyEditForm.tsx

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useState } from 'react'
+
+import { yupResolver } from '@hookform/resolvers/yup'
+import { LoadingButton } from '@mui/lab'
+import { Box, Button, FormControl, InputLabel, MenuItem, Stack, Typography } from '@mui/material'
+import { useTranslation } from 'next-i18next'
+import { useForm, Controller } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { AccountHierarchyFormStyles } from '../AccountHierarchyAddForm/AccountHierarchyAddForm.styles'
+import { KiboSelect } from '@/components/common'
+import { UpdateCustomerB2bAccountParams } from '@/lib/types'
+
+import { B2BAccount } from '@/lib/gql/types'
+
+interface AccountHierarchyEditFormProps {
+  accounts: B2BAccount[]
+  accountToEdit: B2BAccount
+  onSave: (data: UpdateCustomerB2bAccountParams) => void
+  onClose: () => void
+}
+
+const useAccountHierarchySchema = () => {
+  const { t } = useTranslation('common')
+  return yup.object({
+    parentAccount: yup.string().required(t('this-field-is-required')),
+  })
+}
+
+const AccountHierarchyEditForm = (props: AccountHierarchyEditFormProps) => {
+  const { accounts, accountToEdit, onSave, onClose } = props
+
+  const [isLoading, setLoading] = useState<boolean>(false)
+  const [selectedParentAccount, setSelectedParentAccount] = useState<B2BAccount>()
```

Comment:
Instead of holding entire selected account, just set selected account id 
const [selectedParentAccountId, setSelectedParentAccountId] = useState<string>('')
const selectedParentAccount = accounts.filter(account=> .....)

Reply:
selectedParentAccount details are required to show in company name in UI